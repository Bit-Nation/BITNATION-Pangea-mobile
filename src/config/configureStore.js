// @flow

import { createStore, applyMiddleware, compose, type Store } from 'redux';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import createSagaMonitor from '@clarketm/saga-monitor';
import reducers from '../reducers';
import rootSaga from '../sagas';

/**
 * @desc Configures a Redux store.
 * @return {Store} Created store object.
 */
export default function configureStore(): Store {
  const sagaMonitor = createSagaMonitor({
    level: 'log',
    actionDispatch: true,
  });
  const sagaMiddleware = createSagaMiddleware({ sagaMonitor });

  const enhancer = compose(
    applyMiddleware(sagaMiddleware),
    applyMiddleware(logger),
  );

  const store = createStore(reducers, enhancer);
  sagaMiddleware.run(rootSaga);
  return store;
}
