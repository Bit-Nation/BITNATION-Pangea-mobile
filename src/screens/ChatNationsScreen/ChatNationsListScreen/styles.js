// @flow

import { MediaQueryStyleSheet } from 'react-native-responsive';

import GlobalStyles from '../../../global/Styles';
import Colors from '../../../global/colors';

const styles = MediaQueryStyleSheet.create({
  ...GlobalStyles,

  /* Example of over-writing a global style sheet value */
  /*
  tabTextStyle: {
    color: Colors.Red,
  },
  */
});

export default styles;
