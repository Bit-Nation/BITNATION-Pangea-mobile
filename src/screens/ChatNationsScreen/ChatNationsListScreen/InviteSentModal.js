// @flow

import React from 'react';
import {
  View,
  Modal,
  Image,
} from 'react-native';
import { Text } from 'native-base';

import Button from '../../../components/common/Button';
import styles from './styles';
import i18n from '../../../global/i18n';

type Props = {
  /**
   * @desc Function to close the modal
   */
  done: () => void,
  /**
   * @desc Modal visibility
   */
  visible: boolean
};

const InviteSentModal = ({
  done, visible,
}: Props) => (
  <Modal
    animationType='slide'
    transparent
    visible={visible}
  >
    <View style={styles.modalContainer}>
      <View style={[styles.modalContent, styles.inviteSuccessModal]}>
        <View style={styles.profileArea}>
          <Text style={styles.modalTitle}>{i18n.t('screens.chat.inviteSent')}</Text>
          <Text style={styles.modalBody}>
            You have invited Sandy Apples to chat.
          </Text>
        </View>
        <View style={styles.buttonArea}>
          <Button
            enabled
            styleTitle={styles.newAccountText}
            title={i18n.t('common.ok').toUpperCase()}
            onPress={done}
          />
        </View>
      </View>
    </View>
  </Modal>
);

export default InviteSentModal;
