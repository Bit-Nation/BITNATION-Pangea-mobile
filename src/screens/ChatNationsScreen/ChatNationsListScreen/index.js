// @flow

import React from 'react';
import {
  View,
  SectionList,
  Clipboard,
} from 'react-native';
import _ from 'lodash';
import { Fab, Text } from 'native-base';
import ActionSheet from 'react-native-actionsheet';

import BackgroundImage from '../../../components/common/BackgroundImage';
import styles from './styles';
import { screen } from '../../../global/Screens';
import ChatListItem from '../../../components/common/ChatListItem';
import NationListHeader from '../../../components/common/NationListHeader';
import FakeNavigationBar from '../../../components/common/FakeNavigationBar';
import i18n from '../../../global/i18n';
import AssetsImages from '../../../global/AssetsImages';
import type { NationIdType, NationType } from '../../../types/Nation';
import type { Navigator } from '../../../types/ReactNativeNavigation';
import ScreenTitle from '../../../components/common/ScreenTitle';

type Props = {
  /**
   * @desc React Native Navigation navigator object.
   */
  navigator: Navigator,
  /**
   * @desc Selected Tab Name
   */
  selectedTab: string,
  /**
   * @desc List of all nations
   */
  nations: Array<NationType>,
  /**
   * @desc List of all nations
   */
  myNations: Array<NationType>,
  /**
   * @desc Function to be called when an item is selected from the list
   * @param id ID of the nation to be opened
   * @param isBot Flag to indicate if the selected channel is bot
   */
  onSelectItem: (id: number, isBot: boolean) => void,
};

const ChatNationsListScreen = ({
  selectedTab,
  nations,
  myNations,
  onSelectItem,
}: Props) => {
  const nationsToDisplay = selectedTab === 'ALL_NATIONS' ?
    _.filter(nations, nation => nation.idInSmartContract >= 0)
    :
    _.filter(nations, nation => (
      _.indexOf(myNations, nation.id) !== -1) && (nation.idInSmartContract >= 0));
  const sortedNations = _.sortBy(nationsToDisplay, nation => nation.nationName);
  const groups = _.groupBy(sortedNations, nation => nation.nationName.charAt(0));
  let sections = _.map(groups, (group, key) => ({
    title: key,
    data: group,
  }));
  const bots = [{ title: 'Bots', data: [{ name: 'Dr. FreudBot', isBot: true, id: 0 }] }];
  sections = bots.concat(sections);

  return (
    <View style={styles.nationsScreenContainer}>
      <BackgroundImage />
      <FakeNavigationBar />
      <ScreenTitle title={i18n.t('screens.chat.title')} />
      <SectionList
        renderItem={(item) => {
          const nation = item.item;
          if (nation.isBot === true) {
            return (<ChatListItem
              text={nation.name}
              participants=''
              itemIcon={AssetsImages.ChatUI.botIcon}
              onPress={id => onSelectItem(id, true)}
              id={nation.id}
            />);
          }
            return (<ChatListItem
              text={nation.nationName}
              participants=''
              itemIcon={0}
              onPress={id => onSelectItem(id, false)}
              id={nation.id}
            />);
        }}
        keyExtractor={item => item.id}
        renderSectionHeader={({ section }) => <NationListHeader title={section.title} />}
        sections={sections}
        style={styles.sectionList}
      />
    </View>
  );
};

export default ChatNationsListScreen;
