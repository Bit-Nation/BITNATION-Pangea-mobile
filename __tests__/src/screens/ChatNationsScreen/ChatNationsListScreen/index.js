import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import ChatNationsListScreen from '../../../../../src/screens/ChatNationsScreen/ChatNationsListScreen';

describe('ChatNationsListScreen', () => {
  let chatListScreen;

  const initialStateMock = {};
  const storeMock = configureStore([]);
  const propsMock = {
    onSelectItem: jest.fn(),
    selectedTab: 'ALL_NATIONS',
    nations: [],
    myNations: [],
  };

  beforeEach(() => {
    chatListScreen =
      shallow(<ChatNationsListScreen {...propsMock} store={storeMock(initialStateMock)} />);
  });

  test('Rendering', () => {
    expect(chatListScreen).toMatchSnapshot();
  });
});
