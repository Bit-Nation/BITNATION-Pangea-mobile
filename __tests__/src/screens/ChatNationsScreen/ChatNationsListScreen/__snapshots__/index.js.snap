// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatNationsListScreen Rendering 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChatNationsListScreen
    myNations={Array []}
    nations={Array []}
    onSelectItem={[MockFunction]}
    selectedTab="ALL_NATIONS"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <BackgroundImage
          source={1}
          style={undefined}
        />,
        <FakeNavigationBar
          navBarHidden={false}
        />,
        <ScreenTitle
          title="Chat"
        />,
        <SectionList
          data={Array []}
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          renderSectionHeader={[Function]}
          scrollEventThrottle={50}
          sections={
            Array [
              Object {
                "data": Array [
                  Object {
                    "id": 0,
                    "isBot": true,
                    "name": "Dr. FreudBot",
                  },
                ],
                "title": "Bots",
              },
            ]
          }
          stickySectionHeadersEnabled={true}
          style={
            Object {
              "flex": 1,
            }
          }
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />,
      ],
      "style": Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "source": 1,
          "style": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "navBarHidden": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "Chat",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Array [],
          "disableVirtualization": false,
          "horizontal": false,
          "initialNumToRender": 10,
          "keyExtractor": [Function],
          "maxToRenderPerBatch": 10,
          "onEndReachedThreshold": 2,
          "renderItem": [Function],
          "renderSectionHeader": [Function],
          "scrollEventThrottle": 50,
          "sections": Array [
            Object {
              "data": Array [
                Object {
                  "id": 0,
                  "isBot": true,
                  "name": "Dr. FreudBot",
                },
              ],
              "title": "Bots",
            },
          ],
          "stickySectionHeadersEnabled": true,
          "style": Object {
            "flex": 1,
          },
          "updateCellsBatchingPeriod": 50,
          "windowSize": 21,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <BackgroundImage
            source={1}
            style={undefined}
          />,
          <FakeNavigationBar
            navBarHidden={false}
          />,
          <ScreenTitle
            title="Chat"
          />,
          <SectionList
            data={Array []}
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            renderSectionHeader={[Function]}
            scrollEventThrottle={50}
            sections={
              Array [
                Object {
                  "data": Array [
                    Object {
                      "id": 0,
                      "isBot": true,
                      "name": "Dr. FreudBot",
                    },
                  ],
                  "title": "Bots",
                },
              ]
            }
            stickySectionHeadersEnabled={true}
            style={
              Object {
                "flex": 1,
              }
            }
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />,
        ],
        "style": Object {
          "alignItems": "stretch",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-start",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "source": 1,
            "style": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "navBarHidden": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "Chat",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [],
            "disableVirtualization": false,
            "horizontal": false,
            "initialNumToRender": 10,
            "keyExtractor": [Function],
            "maxToRenderPerBatch": 10,
            "onEndReachedThreshold": 2,
            "renderItem": [Function],
            "renderSectionHeader": [Function],
            "scrollEventThrottle": 50,
            "sections": Array [
              Object {
                "data": Array [
                  Object {
                    "id": 0,
                    "isBot": true,
                    "name": "Dr. FreudBot",
                  },
                ],
                "title": "Bots",
              },
            ],
            "stickySectionHeadersEnabled": true,
            "style": Object {
              "flex": 1,
            },
            "updateCellsBatchingPeriod": 50,
            "windowSize": 21,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
